packages/gguf/scripts/generate-llm.ts

```diff
@@ -0,0 +1,179 @@
+	LLM_KV_POOLING_TYPE: "number", // TODO: this is an enum
+	LLM_KV_ATTENTION_CLAMP_KQV: "number",
+	LLM_KV_ATTENTION_MAX_ALIBI_BIAS: "number",
+	LLM_KV_SSM_CONV_KERNEL: "number",
+	LLM_KV_SSM_INNER_SIZE: "number",
+	LLM_KV_SSM_STATE_SIZE: "number",
+	LLM_KV_SSM_TIME_STEP_RANK: "number",
+	LLM_KV_LOGIT_SCALE: "number",
+};
+
+interface Arch {
+	cppConst: string,      // for example: "LLM_ARCH_LLAMA"
+	name: string,          // for example: "llama"
+	tsName: string,        // for example: "ArchLlama"
+	tensorNames: string[], // for example: "token_embd"
+	hparams: string[],
+};
+
+async function main() {
+	const res = await fetch(SOURCE_CPP_URL);
+	const cppSource = await res.text();
+
+	/////////////////////////////////////
+	// extract list of all architectures
+	const archList: Arch[] = [];
+	const matchedArchList = cppSource.match(/LLM_ARCH_NAMES = ([^;]+)/)!![1].split("\n");
+	for (const line of matchedArchList) {
+		const matched = line.match(/(LLM_ARCH_[A-Z0-9_]+),\s+"([^"]+)"/);
+		if (matched && !matched[0].match(/unknown/)) {
+			archList.push({
+				cppConst: matched[1],
+				name: matched[2],
+				tsName: snakeToPascal(matched[1].replace('LLM_', '')),
+				tensorNames: [],
+				hparams: [],
+			});
+		}
+	}
+
+	/////////////////////////////////////
+	// extract map constant name to kv name
+	// for example: LLM_KV_ATTENTION_LAYERNORM_RMS_EPS ==> "%s.attention.layer_norm_rms_epsilon"
+	const constToKVName: { [cppConst: string]: string } = {};
+	const matchedKVList = cppSource.match(/LLM_KV_NAMES = ([^;]+)/)!![1].split("\n");
+	for (const line of matchedKVList) {
+		const matched = line.match(/(LLM_KV_[A-Z0-9_]+)[,\s]+"([^"]+)"/);
+		if (matched) {
+			constToKVName[matched[1]] = matched[2];
+		}
+	}
+
+	/////////////////////////////////////
+	// extract list of tensor names based on architecture
+	// TODO: unused for now
+	const matchedTensorList = cppSource.match(/LLM_TENSOR_NAMES = ([^;]+)/)!![1].split("\n");
+	let currCppConst = '';
+	for (const line of matchedTensorList) {
+		// check if current line has LLM_ARCH_*
+		const cppConst = (line.match(/LLM_ARCH_[A-Z0-9_]+/) || [])[0];
+		if (cppConst) {
+			currCppConst = cppConst;
+			continue;
+		}
+		// check if current line has LLM_TENSOR_*
+		const tensorName = (line.match(/LLM_TENSOR_[A-Z0-9_]+[,\s]+"([^"]+)"/) || [])[1];
+		if (tensorName) {
+			const arch = archList.find(a => a.cppConst === currCppConst)!!;
+			if (arch) arch.tensorNames.push(tensorName);
+		}
+	}
+
+	/////////////////////////////////////
+	// extract list of hyper params based on architecture
+	let insideLoadHParamsFn = false;
+	currCppConst = '';
+	for (const line of cppSource.split("\n")) {
+		// check if current line is function llm_load_hparams()
+		if (line.startsWith("static void llm_load_hparams")) {
+			insideLoadHParamsFn = true;
+		}
+		if (!insideLoadHParamsFn) {
+			continue;
+		}
+		// check if current line has LLM_ARCH_*
+		const cppConst = (line.match(/case (LLM_ARCH_[A-Z0-9_]+)/) || [])[1];
```

Comment:
`LLM_ARCH_[A-Z0-9_]+` part is also used on line 100 https://github.com/ngxson/huggingface.js/blob/5313453c75b3f72fd1381762aa6450d33f77749e/packages/gguf/scripts/generate-llm.ts#L100

therefore, we should declare this regex as const `cont RE_XYZ = /LLM_ARCH_[A-Z0-9_]+)/` & re-use this re in this line like `new Regexp(/case ${RE_XYZ}/)`